core/src/main/java/io/fabric8/maven/core/util/AsciiDocParser.java

```diff
@@ -0,0 +1,156 @@
+     * |===
+     * |Filename
+     * |Kind
+     *
+     * a|ConfigMap
+     * a|`cm`
+     *
+     * |CronJob
+     * |cronjob
+     * |===
+     * </pre>
+     *
+     * Basically the table contains an optional attributes section (@code{[]}), then the headers of the columns.
+     * Between line breaks, both columns representing the kind (first column) and filename (second column).
+     * Notice that one-line columns definition @code{|Cell in column 1, row 1|Cell in column 1, row 1} is not supported.
+     *
+     * This method returns an @code{IllegalArgumentException} if does not contain two columns.
+     *
+     * @param table definition in AsciiDoc format.
+     * @return A serialization of all columns, being pair elements the first column and the odd elements the second column. In previous example @code{"cm","ConfigMap","cronjob","CronJob"}
+     */
+    public Map<String, List<String>> serializeKindFilenameTable(final InputStream table) {
+
+        final Map<String, List<String>> serializedContent = new HashMap<>();
+
+        try (final BufferedReader tableContent = new BufferedReader(new InputStreamReader(table))) {
+
+            skipUntilColumns(tableContent);
+
+            boolean endTable = false;
+
+            while (!endTable) {
+                final List<String> readRow = readRow(tableContent);
+                final String separator = readEmptyLineOrEndTable(tableContent);
+
+                String kind = readRow.get(0);
+                serializedContent.put(kind, readRow.subList(1, readRow.size()));
+
+                if (END_TABLE.equals(separator)) {
+                    endTable = true;
+                }
+
+            }
+
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+
+        return serializedContent;
+
+    }
+
+    private List<String> readRow(final BufferedReader tableContent) throws IOException {
+        final List<String> content = new ArrayList<>();
+
+        final String firstColumn = readColumn(tableContent);
+        final String secondColumn = readColumn(tableContent);
+
+        content.add(firstColumn);
+        final String[] filenameTypes = secondColumn.split(",");
+
+        for (String filenameType : filenameTypes) {
+            content.add(filenameType.trim());
+        }
+
+        return content;
+    }
+
+    private String readColumn(final BufferedReader tableContent) throws IOException {
+        final String column = tableContent.readLine();
+
+        if(column == null || column.isEmpty()) {
+            throw new IllegalArgumentException("Trying to read a column but white line or EOF was found.");
+        }
+
+        int separator;
+        if ((separator = column.indexOf("|")) < 0) {
+            throw new IllegalArgumentException(String.format("Expected the initial of a column with (|) but %s found.", column));
+        }
+
+        return column.trim().substring(separator + 1)
+            .replaceAll("[`_*]", "");
```

Comment:
I would put another trim after the `replaceAll` to catch also things like `* dc*, ..`. Not very likely but I think would be a quick win.